rest/admin_api.go

```diff
@@ -266,6 +267,100 @@ func (h *handler) handleDbOffline() error {
 	return nil
 }
 
+// handleIndexInit allows async index initialization status to be viewed
+func (h *handler) handleGetIndexInit() error {
+	b, err := h.db.AsyncIndexInitManager.GetStatus(h.ctx())
+	if err != nil {
+		return err
+	}
+	h.writeRawJSON(b)
+	return nil
+}
+
+type PostIndexInitRequest struct {
+	NumPartitions *uint32 `json:"num_partitions"`
+}
+
+func (req PostIndexInitRequest) Validate() error {
+	if req.NumPartitions == nil {
+		return base.HTTPErrorf(http.StatusBadRequest, "num_partitions is required")
+	}
+	if *req.NumPartitions < 1 {
+		return base.HTTPErrorf(http.StatusBadRequest, "num_partitions must be greater than 0")
+	}
+	return nil
+}
+
+// handleIndexInit allows async index initialization to be run or managed
+func (h *handler) handlePostIndexInit() error {
+	action := cmp.Or(h.getQuery("action"), "start")
+
+	if action == "stop" {
```

Comment:
What do you think about a set of tests to test validation of input:

- action incorrectly
- calling this function when GSI is not enabled
- bad json
- mismatched partitions

Reply:
agreed, will add